packages/rum/src/domain/record/privacy.ts

```diff
@@ -1,91 +1,386 @@
 import {
-  InputPrivacyMode,
+  NodePrivacyLevel,
+  NodePrivacyLevelInternal,
   PRIVACY_ATTR_NAME,
+  PRIVACY_ATTR_VALUE_ALLOW,
+  PRIVACY_ATTR_VALUE_MASK,
+  PRIVACY_ATTR_VALUE_MASK_FORMS_ONLY,
   PRIVACY_ATTR_VALUE_HIDDEN,
-  PRIVACY_ATTR_VALUE_INPUT_IGNORED,
-  PRIVACY_ATTR_VALUE_INPUT_MASKED,
+  PRIVACY_CLASS_ALLOW,
+  PRIVACY_CLASS_MASK,
+  PRIVACY_CLASS_MASK_FORMS_ONLY,
   PRIVACY_CLASS_HIDDEN,
+  FORM_PRIVATE_TAG_NAMES,
+  CENSORED_STRING_MARK,
+  CENSORED_IMG_MARK,
+  // Deprecated (now aliased) below
   PRIVACY_CLASS_INPUT_IGNORED,
   PRIVACY_CLASS_INPUT_MASKED,
+  PRIVACY_ATTR_VALUE_INPUT_IGNORED,
+  PRIVACY_ATTR_VALUE_INPUT_MASKED,
 } from '../../constants'
 
-// PRIVACY_INPUT_TYPES_TO_IGNORE defines the input types whose input
-// events we want to ignore by default, as they often contain PII.
-// TODO: We might want to differentiate types to fully ignore vs types
-// to obfuscate.
-const PRIVACY_INPUT_TYPES_TO_IGNORE = ['email', 'password', 'tel']
-
-// Returns true if the given DOM node should be hidden. Ancestors
-// are not checked.
-export function nodeShouldBeHidden(node: Node): boolean {
-  return (
-    isElement(node) &&
-    (node.getAttribute(PRIVACY_ATTR_NAME) === PRIVACY_ATTR_VALUE_HIDDEN ||
-      node.classList.contains(PRIVACY_CLASS_HIDDEN))
-  )
-}
+import { makeStylesheetUrlsAbsolute } from './serializationUtils'
 
-// Returns true if the given DOM node should be hidden, recursively
-// checking its ancestors.
-export function nodeOrAncestorsShouldBeHidden(node: Node | null): boolean {
-  if (!node) {
-    return false
-  }
+import { shouldIgnoreElement } from './serialize'
 
-  if (nodeShouldBeHidden(node)) {
-    return true
-  }
+const TEXT_MASKING_CHAR = '᙮'
+const MIN_LEN_TO_MASK = 80
+const WHITESPACE_TEST = /^\s$/
 
-  return nodeOrAncestorsShouldBeHidden(node.parentNode)
+export function getInitialPrivacyLevel(): NodePrivacyLevelInternal {
```

Comment:
`mask-forms-only` is considered an internal state, and gets exposed to the rest of the codebase as `mask` or `allow` depending on the element type. The JS config should accept `mask-forms-only`.

Reply:
Ah, this is confusing. I thought `NodePrivacyLevel` was designed to be public and listed all values accepted by the init configuration, contrary to `NodePrivacyLevelInternal` which would be a superset with internal states. Maybe we can find better names to differentiate both types?

Reply:
I think this point should be deferred and considered with the (currently descoped) init configuration spec since it pivots on the values we expose for customer use.